### DISCUSSION TITLE
Multiple floating elements colliding/overlapping

### DISCUSSION THREAD

**Discussion by matisszemturis** [floating-ui/floating-ui#2602] [Q&A]
Hello!
I want to create Notion/Google Docs like commenting system.
1. If there is free space, each comment is anchored to right-start of a related comment.
2. If there is not enough space, the selected comment is prioritized and the rest are pushed up/down.

Do you have any ideas on how to handle such a use case?

![image](https://github.com/floating-ui/floating-ui/assets/50548714/733fc346-d0f0-4762-8749-5c3c8ea3bb34)

**Comment by atomiks** [comment]
Make the floating element a vertical flexbox column. All comments go inside that: https://codesandbox.io/s/flamboyant-poincare-nr6cr8?file=/src/App.js

Additional logic:

- If comment = 1, use `right-start` placement
- If comment > 1, use `right` placement, so they're centered

**Reply by matisszemturis** [reply]
This is a nice way of handling all items in a single column with equal gaps, but sometimes comments are spread apart and gaps between comments should be bigger. 
If comment has enough space on top/bottom, it should align "right-start" with relevant mark node.
If comment does't have enough space it should shift top/bottom.

**Reply by atomiks** [reply]
Yeah probably requires custom middleware/logic. Issue is currently being tracked already though: https://github.com/floating-ui/floating-ui/issues/2268

**Reply by BrianHung** [reply]
Is there an easier way to combine floating ui with d3 force graph to prevent overlap but also allowing for anchoring?
